src/main/java/roboy/memory/nodes/Roboy.java

```diff
@@ -0,0 +1,132 @@
+package roboy.memory.nodes;
+
+import roboy.dialog.Config;
+import roboy.memory.Neo4jMemory;
+import roboy.memory.Neo4jRelationships;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Encapsulates a MemoryNodeModel and enables dialog states to easily store
+ * and retrieve information about Roboy.
+ */
+public class Roboy {
+    private MemoryNodeModel roboy;
+    Neo4jMemory memory;
+    // Memory is not queried in NOROS mode.
+    private boolean memoryROS;
+
+    /**
+     * Initializer for the Roboy node
+     */
+    public Roboy() {
+        this.roboy = new MemoryNodeModel(true);
+        this.memory = Neo4jMemory.getInstance();
+        this.memoryROS = Config.MEMORY;
+        this.InitializeRoboy("roboy"); // May be "roboy junior"
+    }
+
+    /**
+     * This method initializes the roboy property as a node that
+     * is in sync with memory and represents the Roboy itself.
+     *
+     * If something  goes wrong during querying, Roboy stays empty
+     * and soulless, and has to fallback
+     */
+    // TODO consider a fallback for the amnesia mode
+    private void InitializeRoboy(String name) {
+        roboy.setProperty("name", name);
+        roboy.setLabel("Robot");
+
+        if(memoryROS) {
```

Comment:
memory needs to be cut down on the DialogSystem.java level, not in here.
this is a very good example why singletons are bad.
please get rid of this kind of checks

Reply:
This checks are provided through the Config class as a flag to let the nodes know if they are supposed to access/flush the data to the Memory or leave it intact for "no memory" configuration testing purposes (that was done before me)

Reply:
Regarding the memory singleton. Currently, it is somewhat difficult to refactor all the places in the Dialog to get rid of it and ensure it works the same. I suppose, we may approach this in the new Dialog State Machine from scratch?